nic_operator/nic_operator_ci_test.sh

```diff
@@ -466,6 +489,99 @@ function test_predefined_name {
     return 0
 }
 
+function test_host_device {
+    local resource_name_prefix="nvidia"
+    local resource_name="hostdev"
+
+    status=0
+
+    echo "Testing Host device..."
+    echo ""
+
+    sudo apt-get install -y rdma-core
+
+    load_core_drivers
+    sleep 2
+
+    load_rdma_modules
+    sleep 2
+
+    local sample_file="$ARTIFACTS"/host-device-nic-cluster-policy.yaml
+
+    configure_common "$sample_file"
+    configure_host_device "$sample_file" "$resource_name_prefix" "$resource_name"
+
+    nic_policy_create "$sample_file"
+    let status=$status+$?
+    if [[ "$status" != "0" ]]; then
+        echo "Error: error in creating the example nic_policy."
+        return $status
+    fi
+
+    sleep 10
+
+    test_rdma_single_pod "" "$MACVLAN_NETWORK_DEFAULT_NAME" "${resource_name_prefix}/${resource_name}"
+    let status=$status+$?
+    if [[ "$status" != "0" ]]; then
+        echo "Error: host device plugin testing Failed!"
+        return $status
+    fi
+
+    delete_nic_cluster_policies
+    let status=$status+$?
+    if [[ "$status" != "0" ]]; then
+        echo "Error: Couldn't delete $sample_file!"
+        return $status
+    fi
+    return 0
+}
+
+function test_ofed_and_host_device {
+    local resource_name_prefix="nvidia"
+    local resource_name="hostdev"
+
+    status=0
+
+    echo "Testing OFED and Host device..."
+    echo ""
+
+    local sample_file="$ARTIFACTS"/ofed-and-host-device-nic-cluster-policy.yaml
+
+    configure_common "$sample_file"
+    configure_ofed "$sample_file"
+    configure_host_device "$sample_file" "$resource_name_prefix" "$resource_name"
+
+    nic_policy_create "$sample_file"
+    let status=$status+$?
+    if [[ "$status" != 0 ]]; then
+        echo "Error: error in creating the example nic_policy."
+        return $status
+    fi
+
+    sleep 10
+
+    test_ofed_drivers
+    let status=$status+$?
+    if [[ "$status" != "0" ]]; then
+        echo "Error: Ofed modules failed!"
+        return $status
+    fi
+
+    test_rdma_single_pod "" "$MACVLAN_NETWORK_DEFAULT_NAME" "${resource_name_prefix}/${resource_name}"
```

Comment:
Do we need MACVLAN_NETWORK_DEFAULT_NAME here?

Reply:
fixed it to use a hostdevice network